src/senselab/audio/tasks/preprocessing/preprocessing.py

```diff
@@ -103,40 +103,111 @@ def select_channel_from_audios(audios: List[Audio], channel_index: int) -> List[
     return mono_channel_audios
 
 
-def chunk_audios(data: List[Tuple[Audio, Tuple[float, float]]]) -> List[Audio]:
-    """Chunks the input audios based on the start and end timestamp.
+def extract_segments(data: List[Tuple[Audio, List[Tuple[float, float]]]]) -> List[List[Audio]]:
+    """Extracts segments from an audio file.
 
     Args:
-        data: List of tuples containing an Audio object and a tuple with start and end (in seconds) for chunking.
+        data: List of tuples containing an Audio object and a list of tuples with start
+            and end (in seconds) for chunking.
 
     Returns:
-        List of Audios that have been chunked based on the provided timestamps
+        List of lists of Audios that have been chunked based on the provided timestamps.
     """
-    chunked_audios = []
-
+    extracted_segments = []
     for audio, timestamps in data:
-        start, end = timestamps
-        if start < 0:
-            raise ValueError("Start time must be greater than or equal to 0.")
-        duration = audio.waveform.shape[1] / audio.sampling_rate
-        if end > duration:
-            raise ValueError(f"End time must be less than the duration of the audio file ({duration} seconds).")
-        start_sample = int(start * audio.sampling_rate)
-        end_sample = int(end * audio.sampling_rate)
-        chunked_waveform = audio.waveform[:, start_sample:end_sample]
-
-        chunked_audios.append(
-            Audio(
-                waveform=chunked_waveform,
-                sampling_rate=audio.sampling_rate,
-                metadata=audio.metadata.copy(),
-                orig_path_or_id=audio.orig_path_or_id,
+        single_audio_segments = []
+        for start, end in timestamps:
+            if start < 0:
+                raise ValueError("Start time must be greater than or equal to 0.")
+            duration = audio.waveform.shape[1] / audio.sampling_rate
+            if end > duration:
+                raise ValueError(f"End time must be less than the duration of the audio file ({duration} seconds).")
+            start_sample = int(start * audio.sampling_rate)
+            end_sample = int(end * audio.sampling_rate)
+            chunked_waveform = audio.waveform[:, start_sample:end_sample]
+
+            single_audio_segments.append(
+                Audio(
+                    waveform=chunked_waveform,
+                    sampling_rate=audio.sampling_rate,
+                    metadata=audio.metadata.copy(),
+                    orig_path_or_id=audio.orig_path_or_id,
+                )
             )
+        extracted_segments.append(single_audio_segments)
+    return extracted_segments
+
+
+def pad_audios(audios: List[Audio], desired_samples: int) -> List[Audio]:
```

Comment:
@ibevers does this work with stereo audio recordings?

Reply:
Yes, `padded_waveform = torch.nn.functional.pad(audio.waveform, (0, padding_needed))` pads along the time dimension, so each channel of the audio is padded. I updated the test for this function to test for this.